test/new-e2e/tests/remote-config/utils_test.go

```diff
@@ -0,0 +1,38 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package remoteconfig
+
+import (
+	_ "embed"
+	"fmt"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+// assertLogsEventually will verify that a given `agentName` component's logs contain a pattern.
+// It will continually retry until the `expectedLogPattern` is found or the `maxRetries` is reached,
+// waiting `retryInterval` between each attempt.
+// If the `expectedLogPattern` is not found or an error occurs, the calling test will fail.
+func assertLogsEventually(t *testing.T, rh *components.RemoteHost, agentName string, expectedLogPattern string, waitFor time.Duration, tick time.Duration) {
+	assert.EventuallyWithTf(t, func(c *assert.CollectT) {
+		output, err := rh.Execute(fmt.Sprintf("cat /var/log/datadog/%s.log", agentName))
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
I'd like to find a way to do this more efficiently, adding a card on our side to investigate handling streams reading remote files